src/peer_manager.rs

```diff
@@ -1215,21 +1213,27 @@ impl PeerManager {
         let _ = self.set_state(peer_id, PeerState::Routing(RoutingConnection::Direct));
     }
 
-    /// Returns the `NUM_TUNNEL_VIA_NODES` closest peers from our section which can be potential
-    /// tunnel node.
-    pub fn potential_tunnel_nodes(&self) -> Vec<(XorName, PeerId)> {
-        let our_section = self.routing_table
-            .other_close_names(self.our_public_id.name())
-            .unwrap_or_default();
-        self.peer_map
-            .peers()
-            .filter_map(|peer| if our_section.contains(peer.name()) &&
-                                  peer.state.can_tunnel_for() {
-                peer.peer_id.map_or(None, |peer_id| Some((*peer.name(), peer_id)))
-            } else {
-                None
+    /// Returns the closest section to the peer and the closest section to own.
+    pub fn potential_tunnel_nodes(&self, name: &XorName) -> Vec<(XorName, PeerId)> {
+        self.routing_table
+            .our_section()
+            .iter()
+            .chain(self.routing_table.closest_section(name).1.iter())
```

Comment:
I think we should use `self.routing_table.get_section(name)` here to ensure we return the right section (this will return a different section if the one for `name` is still empty) and to avoid exposing another RT function.